tket/src/Predicates/include/Predicates/Predicates.hpp

```diff
@@ -238,6 +238,18 @@ class NoBarriersPredicate : public Predicate {
   std::string to_string() const override;
 };
 
+/**
+ * Asserts that any internal measurements can be commuted to the end of the
+ * circuit
+ */
+class CommutableMeasuresPredicate : public Predicate {
+ public:
+  bool verify(const Circuit& circ) const override;
+  bool implies(const Predicate& other) const override;
+  PredicatePtr meet(const Predicate& other) const override;
+  std::string to_string() const override;
+};
+
 /**
  * Asserts that any measurements occur at the end of the circuit
  */
```

Comment:
Making `NoMidMeasurePredicate` inherit from `CommutableMeasuresPredicate` here breaks the serialization tests. I could not find the reason for that.